### PR TITLE
Adjustments for configure --enable-rexgen

### PR DESCRIPTION
**Pull request by frank-dittrich**
Adjust john to "recent" rexgen changes (including some interface
changes), point out known issues and possibility that rexgen support
might have to be dropped in future releases

### CODE REVIEW

File: doc/README.librexgen
Comment (solardiz) [inline doc/README.librexgen]: You probably meant `invalid`, unless you deliberately invalidated this word. ;-)

File: src/configure
Comment (solardiz) [inline src/configure]: Maybe replace `experimental!` with `unreliable`. This isn't really an experiment for some upcoming feature - it's a feature that's been around for a long time and never became reliable and isn't expected to become reliable now.

File: src/configure.ac
Comment (solardiz) [inline src/configure.ac]: Ditto.

**Review by solardiz — APPROVED** [review]
Looks good to me. Feel free to make the minor edits I suggested or ask for this to be merged as-is. Thank you!

**Comment by solardiz** [comment]
Thanks, @frank-dittrich! Ready to merge now?

**Comment by frank-dittrich** [comment]
@solardiz I fixed the misspelled word, replaced "experimental" with "unreliable" (In my initial commit I was even thinking about "b0rken"), and added another illegal regex causing a segfault.

IMHO ready to merge.
Or should I try to add workarounds in regex.c for known problematic regular expressions like empty `[]` and illegal back references? At least the `[]` should be trivial.
I don't like to add clumsy workarounds in john's regex.c, but I definitely wouldn't want to become a rexgen maintainer.
May be workarounds for broken rexgen code should be separate commits so that they can be reverted should rexgen fix these issues in the future.

**Comment by frank-dittrich** [comment]
Just found another segfault:
```
rexgen '[äöü]'
```
Should I add this to doc/README.librexgen?

**Comment by frank-dittrich** [comment]
@magnumripper @solardiz 
Please don't merge. I found a more stable, less troublesome rexgen version, will submit changes after I had some sleep

**Pull request closed without merging**
